extensions/namespace-lookup-management/src/main/java/io/druid/metadata/MetadataNamespaceManager.java

```diff
@@ -0,0 +1,346 @@
+
+  @ManageLifecycle
+  public static class SQLMetadataNamespaceManager implements MetadataNamespaceManager
+  {
+    private static final Logger log = new Logger(SQLMetadataNamespaceManager.class);
+    private final ObjectMapper jsonMapper;
+    private final Supplier<MetadataNamespaceManagerConfig> config;
+    private final Supplier<NamespaceMetadataStorageTablesConfig> dbTables;
+    private final SQLMetadataConnector connector;
+    private final Object lock = new Object();
+    private volatile boolean started = false;
+    private final AtomicBoolean loadedOnce = new AtomicBoolean(false);
+    private final AtomicReference<Map<String, Map<String, Object>>> knownNamespaces = new AtomicReference<Map<String, Map<String, Object>>>(
+        new ConcurrentHashMap<String, Map<String, Object>>()
+    );
+    private ListeningScheduledExecutorService exec;
+    private ListenableFuture<?> future = null;
+    private AtomicReference<Runnable> postPollRunnable = new AtomicReference<>(null);
+
+    @Inject
+    public SQLMetadataNamespaceManager(
+        @Json ObjectMapper jsonMapper,
+        Supplier<MetadataNamespaceManagerConfig> config,
+        Supplier<NamespaceMetadataStorageTablesConfig> dbTables,
+        SQLMetadataConnector connector
+    )
+    {
+      this.jsonMapper = jsonMapper;
+      this.config = config;
+      this.dbTables = dbTables;
+      this.connector = connector;
+      this.exec = createExecService();
+    }
+
+    private static final ListeningScheduledExecutorService createExecService()
+    {
+      return MoreExecutors.listeningDecorator(Execs.scheduledSingleThreaded("DatabaseSegmentManager-Exec--%d"));
+    }
+
+    @Override
+    public void start()
+    {
+      synchronized (lock) {
+        if (started) {
+          return;
+        }
+
+        future = exec.scheduleWithFixedDelay(
+            new Runnable()
+            {
+              @Override
+              public void run()
+              {
+                try {
+                  poll();
+                  final Runnable runnable = postPollRunnable.get();
+                  if (runnable != null) {
+                    runnable.run();
+                  }
+                }catch (Exception e){
+                  log.error(e, "Error in polling for namespace metadata. Sleeping for [%s]", config.get().getPollDuration());
+                }
+              }
+            },
+            0,
+            config.get().getPollDuration().toStandardDuration().getMillis(),
+            TimeUnit.MILLISECONDS
+        );
+
+        started = true;
+      }
+    }
+
+    @Override
+    public void stop()
+    {
+      synchronized (lock) {
+        if (!started) {
+          return;
+        }
+
+        if (future != null) {
+          if (!future.isDone() && !future.cancel(true)) {
+            log.warn("Unable to cancel future! Trying to restart entire executor service");
+            exec.shutdownNow();
+            exec = createExecService();
```

Comment:
would it leave the executor running even after calling stop(), 
wondering if we should always shutdown the exec on stop and create one on start ?